qa/L0_config/check_results.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import yaml
+import os
+import sys
+
+sys.path.insert(0, '../')
+from common.util import check_csv_table_row_column
+
+
+class TestOutputValidator:
+    """
+    Functions that validate the output
+    of the test
+    """
+
+    def __init__(self, config, analyzer_log, server_metrics_file,
+                 model_metrics_GPU_file, model_metrics_inference_file,
+                 test_output_num_rows, num_gpus):
+        self._config = config
+        self._analyzer_log = analyzer_log
+        self._server_metrics_file = server_metrics_file
+        self._model_metrics_GPU_file = model_metrics_GPU_file
+        self._model_metrics_inference_file = model_metrics_inference_file
+        self._test_output_num_rows = test_output_num_rows
+        self._num_gpus = num_gpus
+
+        check_functions = [
+            self.__getattribute__(name)
+            for name in dir(self)
+            if name.startswith('check')
+        ]
+
+        passed_test = True
+        for check_function in check_functions:
+            passed_test &= check_function()
+
+        if passed_test:
+            sys.exit(0)
+        else:
+            sys.exit(1)
+
+    def check_config_test(self):
+        """
+        Check outputs of log and csv tables
+        """
+        metrics_num_columns = 11
+        inference_num_columns = 10
+        server_metrics_num_columns = 5
+        output_tag = "Model"
+
+        passed = check_csv_table_row_column(
+            os.path.abspath(self._server_metrics_file),
+            server_metrics_num_columns, 1 * self._num_gpus, output_tag)
+        passed &= check_csv_table_row_column(
+            os.path.abspath(self._model_metrics_GPU_file), metrics_num_columns,
+            self._test_output_num_rows * self._num_gpus, output_tag)
+        passed &= check_csv_table_row_column(
+            os.path.abspath(self._model_metrics_inference_file),
+            inference_num_columns, self._test_output_num_rows, output_tag)
```

Comment:
You should only use `check_test_results` instead of all above separate checks. I described the `check_test_results` in the `qa/common`.

Reply:
Refactored